src/equations/ideal_glm_mhd_3d.jl

```diff
@@ -397,6 +510,107 @@ Calculate minimum and maximum wave speeds for HLL-type fluxes as in
 end
 
 
+# Very naive way to approximate the edges of the Riemann fan in the normal direction
+@inline function min_max_speed_naive(u_ll, u_rr, normal_direction::AbstractVector,
+                                     equations::IdealGlmMhdEquations3D)
+  rho_ll, rho_v1_ll, rho_v2_ll, rho_v3_ll, _, _, _, _, _ = u_ll
+  rho_rr, rho_v1_rr, rho_v2_rr, rho_v3_rr, _, _, _, _, _ = u_rr
+
+  # Calculate primitive velocity variables
+  v1_ll = rho_v1_ll/rho_ll
+  v2_ll = rho_v2_ll/rho_ll
+  v3_ll = rho_v3_ll/rho_ll
+
+  v1_rr = rho_v1_rr/rho_rr
+  v2_rr = rho_v2_rr/rho_rr
+  v3_rr = rho_v3_rr/rho_rr
+
+  # Compute wave speed estimates in each direction. Requires rotation because
+  # the fast magnetoacoustic wave speed has a nonlinear dependence on the direction
+  norm_ = norm(normal_direction)
+  # Normalize the vector without using `normalize` since we need to multiply by the `norm_` later
+  normal_vector = normal_direction / norm_
+  # Some vector that can't be identical to normal_vector (unless normal_vector == 0)
+  tangent1 = SVector(normal_direction[2], normal_direction[3], -normal_direction[1])
+  # Orthogonal projection
+  tangent1 -= dot(normal_vector, tangent1) * normal_vector
+  tangent1 = normalize(tangent1)
+
+  # Third orthogonal vector
+  tangent2 = normalize(cross(normal_direction, tangent1))
+
+  # Compute the rotated velocities and wave speeds
+  v_normal_ll = v1_ll*normal_vector[1] + v2_ll*normal_vector[2] + v3_ll*normal_vector[3]
+  v_normal_rr = v1_rr*normal_vector[1] + v2_rr*normal_vector[2] + v3_rr*normal_vector[3]
+
+  u_ll_rotated = rotate_to_x(u_ll, normal_vector, tangent1, tangent2, equations)
+  u_rr_rotated = rotate_to_x(u_rr, normal_vector, tangent1, tangent2, equations)
+
+  c_f_ll_rotated = calc_fast_wavespeed(u_ll_rotated, 1, equations)
+  c_f_rr_rotated = calc_fast_wavespeed(u_rr_rotated, 1, equations)
+  v_roe_rotated, c_f_roe_rotated = calc_fast_wavespeed_roe(u_ll_rotated, u_rr_rotated, 1, equations)
+
+  # Estimate the min/max eigenvalues in the normal direction
+  λ_min = min(v_normal_ll - c_f_ll_rotated, v_roe_rotated - c_f_roe_rotated) * norm_
+  λ_max = max(v_normal_rr + c_f_rr_rotated, v_roe_rotated + c_f_roe_rotated) * norm_
+
+  return λ_min, λ_max
+end
+
+
+# Rotate normal vector to x-axis; normal, tangent1 and tangent2 need to be orthonormal
+# Called inside `FluxRotated` in `numerical_fluxes.jl` so the directions
+# has been normalized prior to this rotation of the state vector
+# Note, for ideal GLM-MHD only the velocities and magnetic field variables rotate
+@inline function rotate_to_x(u, normal_vector, tangent1, tangent2, equations::IdealGlmMhdEquations3D)
+  # Multiply with [ 1   0        0       0   0   0        0       0   0;
```

Comment:
Dumb question: Isn't it "multiply *by*"?

Reply:
Either "with" or "by" they are both okay English

Reply:
Thanks!